src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksNewVendorRequestsServiceImpl.java

```diff
@@ -0,0 +1,412 @@
+    protected PaymentWorksIsoFipsCountryDao paymentWorksIsoFipsCountryDao;
+    
+    protected Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap = null; 
+    protected Map<String, SupplierDiversity> paymentWorksToKfsDiversityMap = null;
+    
+    @Override
+    public void createKfsVendorsFromPmwNewVendorRequests() {
+        LOG.info("createKfsVendorsFromPmwNewVendorRequests: was invoked");
+        List<String> pmwNewVendorIdentifers = getPaymentWorksWebServiceCallsService().obtainPmwIdentifiersForPendingNewVendorRequests();
+        if (pmwNewVendorIdentifers.isEmpty()) {
+            LOG.info("createKfsVendorsFromPmwNewVendorRequests: No PENDING New Vendors found to process.");
+            getPaymentWorksNewVendorRequestsReportService().sendEmailThatNoDataWasFoundToProcess();
+        } 
+        else {
+            LOG.info("createKfsVendorsFromPmwNewVendorRequests: Found " + pmwNewVendorIdentifers.size() + " PENDING New Vendors to process.");
+            processEachPaymentWorksNewVendorRequestIntoKFS(pmwNewVendorIdentifers);
+        }
+        LOG.info("createKfsVendorsFromPmwNewVendorRequests: leaving method");
+    }
+    
+    private void processEachPaymentWorksNewVendorRequestIntoKFS(List<String> identifiersForPendingNewVendorsToProcess) {
+        PaymentWorksNewVendorRequestsBatchReportData reportData = new PaymentWorksNewVendorRequestsBatchReportData();
+        reportData.getPendingNewVendorsFoundInPmw().setRecordCount(identifiersForPendingNewVendorsToProcess.size());
+
+        for (String pmwNewVendorRequestId : identifiersForPendingNewVendorsToProcess) {
+            LOG.info("processEachPaymentWorksNewVendorRequestIntoKFS: Processing started for PMW-vendor-id=" + pmwNewVendorRequestId);
+            PaymentWorksVendor stgNewVendorRequestDetailToProcess = getPaymentWorksWebServiceCallsService().obtainPmwNewVendorRequestDetailForPmwIdentifier(pmwNewVendorRequestId, reportData);
+            if (canPaymentWorksNewVendorRequestProcessingContinueForVendor(stgNewVendorRequestDetailToProcess, reportData)) {
+                PaymentWorksVendor savedStgNewVendorRequestDetailToProcess = pmwPendingNewVendorRequestInitialSaveToKfsStagingTable(stgNewVendorRequestDetailToProcess);
+                if (pmwNewVendorSaveToStagingTableWasSuccessful(savedStgNewVendorRequestDetailToProcess, reportData)) {
+                    if (pmwNewVendorRequestProcessingIntoKfsWasSuccessful(savedStgNewVendorRequestDetailToProcess, reportData)) {
+                        getPaymentWorksWebServiceCallsService().sendApprovedStatusToPaymentWorksForNewVendor(savedStgNewVendorRequestDetailToProcess.getPmwVendorRequestId());
+                    }
+                    else {
+                        getPaymentWorksWebServiceCallsService().sendRejectedStatusToPaymentWorksForNewVendor(savedStgNewVendorRequestDetailToProcess.getPmwVendorRequestId());
```

Comment:
It might make sense to have a newVendorSuccessful boolean to keep track of whether the vendor was accepted or rejected. Then you only need to call the sendRejectedStatusToPaymentWorksForNewVendor method once and the code for this method will be much shorter.